include/net/nrf91_lwm2m_app.rst

```diff
@@ -0,0 +1,195 @@
+
+.. figure:: /images/lib_nrf91_lwm2m_app.png
+      :alt: nRF91 LwM2M application software stack
+
+Configuration and implementation
+********************************
+The nRF91 LwM2M client application library is used with the :ref:`lwm2m_client` sample. The sample could be used as a
+starting for creating your own application.
+
+By default, the library uses LTE-M for connecting and doesn't utilize a bootstrap server. LwM2M Queue mode is not in use.
+To use NB-IoT, a bootstrap server, or the queue mode please refer to the forementioned :ref:`lwm2m_client` sample to
+see how these features can be taken into use.
+
+:option:`CONFIG_NRF91_LWM2M_APP` enabled
+   Enables this library
+
+Support for following objects can be set individually but are enabled by default
+
+:option:`CONFIG_NRF91_LWM2M_APP_DEVICE_OBJ_SUPPORT` enabled
+   Enables support for the Device object. Disable only if you're going to implement `reboot` resource on your
+   application cause it's deemed mandatory.
+
+:option:`CONFIG_NRF91_LWM2M_APP_SECURITY_OBJ_SUPPORT` enabled
+   Enables support for the Security object
+
+:option:`CONFIG_NRF91_LWM2M_APP_CONN_MON_OBJ_SUPPORT` enabled
+   Enables support for the Connection Monitor object
+
+:option:`CONFIG_NRF91_LWM2M_APP_LOCATION_OBJ_SUPPORT` enabled
+   Enables support for the Location object
+
+:option:`CONFIG_NRF91_LWM2M_APP_FIRMWARE_UPDATE_OBJ_SUPPORT` enabled
+   Enables support for the Firmware Update object
+
+Defining your own resources
+***************************
+
+This section will walk through a process of creating a custom LwM2M resource. For most application, the
+business logic may be implemented inside custom resources, which are either proprietary to this application
+or may be following some external specification, like IPSO objects. Either way, they extend the resource tree
+by defining own objects on top of what LwM2M already defines.
+
+Before defining a proprietary resource, it is a good practice to check whether similar functionality is
+already defined in `LwM2M Object and Resource Registry`_.
+If there is already such definition you might want to use that, instead of defining your own.
+
+Please note that Zephyr's LwM2M library has only a limited support for some IPSO objects, so extending beyound those,
+requires changes to the internal engine as well.
+
+For this example, we are going to demonstrate how do we define a resource that follow Generic Sensor definition from IPSO.
+First, we look for "Generic Sensor" from the `LwM2M Object and Resource Registry`_ list
+which would give us that the resource object ID would be `3300`. Within that list, we could open a `raw XML`_ file
+or open a bit more human readable view from `the LwM2M editor`_ to see what resources are defined under the object ID 3300.
+
+.. _LwM2M Object and Resource Registry: http://openmobilealliance.org/wp/OMNA/LwM2M/LwM2MRegistry.html
```

Comment:
The links should rather end up in `links.txt`.

Reply:
Seems to be fixed.